plugins/node/opentelemetry-instrumentation-redis-4/src/instrumentation.ts

```diff
@@ -332,6 +338,52 @@ export class RedisInstrumentation extends InstrumentationBase<any> {
     };
   }
 
+  private _getPatchedClientConnect() {
+    const plugin = this;
+    return function connectWrapper(original: Function) {
+      return function patchedConnect(this: any): Promise<void> {
+        const options = this.options;
+
+        const attributes = {
+          [SemanticAttributes.DB_SYSTEM]: DbSystemValues.REDIS,
+          [SemanticAttributes.NET_PEER_NAME]: options?.socket?.host,
+          [SemanticAttributes.NET_PEER_PORT]: options?.socket?.port,
+          [SemanticAttributes.DB_CONNECTION_STRING]: options?.url,
+        };
+
+        const span = plugin.tracer.startSpan(
+          `${RedisInstrumentation.COMPONENT}.connect`,
```

Comment:
Instrumentation currently builds spans names with format:
```js
`${RedisInstrumentation.COMPONENT}-${commandName}`
```

Do we want to stay consistent here and use `-`? I am ok with changing it to dot in all places but it's a bit weird to have a mix from the same instrumentation IMO.

Reply:
I was doubting whether to use a `.` or `-` here (or something else). The actual redis commands have `-`, so I wanted to differentiate between a redis command and connect operation (which isn't a command). But I agree it looks somewhat inconsistent  :thinking:

Reply:
Actually, I am not even sure if the `redis` prefix is encouraged or not [in semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md#semantic-conventions-for-database-client-calls)

> The span name SHOULD be set to a low cardinality value representing the statement executed on the database. It MAY be a stored procedure name (without arguments), DB statement without variable arguments, operation name, etc

[The examples](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md#redis) also suggest naming redis spans in like `"HMSET myhash"`.

Anyway, this is not very relevant to this PR, so feel free to do what you think is best. We can revisit this in the future.

Reply:
For consistency I changed it to `-`, but indeed this might need another PR